tensorstore/internal/path.cc

```diff
@@ -147,7 +147,7 @@ constexpr AsciiSet kUriPathUnreservedChars{
     "abcdefghijklmnopqrstuvwxyz"
     "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
     "0123456789"
-    "-_.!~*'():@&=+$,;/"};
```

Comment:
This was intended to be: https://datatracker.ietf.org/doc/html/rfc2396#section-3.3

S3 doesn't allow $?

Reply:
Actually for S3 signature calculations these are the unreserved characters:
https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html

'A'-'Z', 'a'-'z', '0'-'9', '-', '.', '_', and '~'.

As recommended there, we should just have a separate AsciiSet to use for S3 signatures.

Reply:
Apologies, this (and the test change) should be reverted. The reason for this was to get one of the AWS S3 Signature cases working. In the PUT example, the following:

```
s3://bucket/test$file.text
```

needs to convert to the following in the CanonicalRequest.

```
PUT
/test%24file.text
```


> As recommended there, we should just have a separate AsciiSet to use for S3 signatures.

Will do.